### ISSUE TITLE
sudden size increase due to a new thing called 'security-agent'

### ISSUE THREAD

**Issue by breathe** [DataDog/heroku-buildpack-datadog#204]
Hey -- something changed in this buildpack in the last couple of hours that significantly increases the size foot-print in heroku slug.

I'm now seeing something large called 'security-agent' in the slug that was not present a few hours prior.

`.apt/opt/datadog-agent/embedded/bin/security-agent`

Is this required?  Is there a way to exclude it?  

We don't want any space used in our slugs that we don't need.

**Comment by breathe** [comment]
There is also a new very large file called:

```
/Users/ncohen/Downloads/slugs/itsacheckmate-production/app/.apt/opt/datadog-agent/embedded/lib/libbcc.so.128-NOTFOUND
```

Is that necessary inside the heroku datadog distro?

**Comment by breathe** [comment]
Worried about the size of this agent -- this latest release has gotten _really_ problematically large ..

**Comment by breathe** [comment]
Another weird thing -- why are there .jar files in the distro?  As far as I know there is no java runtime embedded -- so ... its not needed right?

`.apt/opt/datadog-agent/bin/agent/dist/jmx/jmxfetch.jar`

**Comment by klette** [comment]
This is worrying. I haven't checked our slug size the last couple of days, but we're really close to the limit due to the embedded Python already. A sudden 55MB increase would put us over the hard limit, and I would have no choice but to disable datadog at that point.

**Comment by jeremy-lq** [comment]
@breathe Thanks for raising this. slug size is something we're very cognizant of. We'll work on a new buildpack release that treats the security-agent the same as APM/process (ie. remove it if it's not enabled) and take a closer look at JMXfetch. In the interim you can pin the agent version to 6.20.2 / 7.20.2 (or the buildpack to 1.18), as long as you don't require any changes in more recent versions. /cc @arapulido

**Comment by arapulido** [comment]
We have removed the security-agent (that is not needed in Heroku) and some unused dependencies and have released a new version of the buildpack (1.20) with those changes. Thanks again for reporting!

**Comment by arapulido** [comment]
We have also added some tests that will warns us when the slug size has a sudden increase in size
